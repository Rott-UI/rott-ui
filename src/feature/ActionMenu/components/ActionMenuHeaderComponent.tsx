// React Imports
import React, {FC} from 'react'

// Constant Imports
import {COLOURS} from '../../../constants'

// Util and Lib Imports
import {Item, Label} from '../..'

interface ActionMenuHeaderProps {
  title?: Nullable<string>
  subTitle?: Nullable<string>
}

export const ActionMenuHeaderComponent: FC<ActionMenuHeaderProps> = ({title, subTitle}) => {
  return (
    <Item size='full' alignItemsCenter>
      <Item testID='action-menu-header-test-id' width={342}>
        {(title || subTitle) && (
          <Item
            justifyContentCenter
            alignItemsCenter
            size='full'
            backgroundColor={COLOURS.GREY100}
            borderTopStartRadius={12}
            borderTopEndRadius={12}
            height={54}>
            {title && (
              <Label
                fontSize='xl'
                variant='grey-900'
                fontWeight={700}
                marginBottom={subTitle ? 4 : undefined}>
                {title}
              </Label>
            )}
            {subTitle && (
              <Label fontSize='sm' variant='grey-900' fontWeight={500}>
                {subTitle}
              </Label>
            )}
          </Item>
        )}
      </Item>
    </Item>
  )
}
