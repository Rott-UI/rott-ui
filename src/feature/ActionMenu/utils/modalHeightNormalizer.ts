import {display} from '../../../utils'

export const modalHeightNormalizer = (
  dataLength: number,
  titleExist: boolean,
  subTitleExist: boolean,
  separatorTotalHeight: number,
  maxItem: number,
  itemHeight: number
) => {
  const showItemCount = dataLength > maxItem ? maxItem : dataLength
  const headerHeight = display.px(56)
  const listBottomPosition = display.px(93)

  return (
    ((showItemCount * itemHeight +
      (titleExist || subTitleExist ? headerHeight : 0) +
      listBottomPosition) *
      100) /
      display.percentage(100, 'height') +
    separatorTotalHeight
  )
}
