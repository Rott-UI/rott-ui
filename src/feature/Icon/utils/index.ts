// TODO: Icon size converter implementasyonu yapılması gerekmektedir.
