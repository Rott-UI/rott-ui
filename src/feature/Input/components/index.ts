export * from './Input'
export * from './DefaultInput'
export * from './NumericInput'
export * from './PasswordInput'
export * from './IbanInput'
export * from './PhoneInput'
export * from './CVCInput'
export * from './ExpireDateInput'
export * from './CreditCardInput'
export * from './PinPasswordInput'
export * from './DateInput'
export * from './PlateNumberInput'
export * from './AmountInput'
export * from './CheckBoxInput'
export * from './SelectInput'
export * from './InputValidation'
