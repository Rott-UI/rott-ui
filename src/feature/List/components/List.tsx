// React Imports
import {FC, forwardRef} from 'react'

// React Native Imports
import {StyleSheet} from 'react-native'

// Component Imports
import {ListStyles} from '../styles'
import {CommonUiProps, Variant} from '../../models'
import {Item} from '../../Item'

// Package Components
import {FlashList, FlashListProps} from '@shopify/flash-list'
import {Separator} from '../../Separator'

interface ListProps<T> extends CommonUiProps, FlashListProps<T> {
  renderSeparator?: boolean
  separatorWidth?: number
  separatorHeight?: number
  separatorVariant?: Variant
  headerSeparator?: boolean
  footerSeparator?: boolean
  ref?: any
}

export const List: FC<ListProps<any>> = forwardRef(
  (
    {
      renderSeparator,
      data,
      width = '100%',
      estimatedItemSize = 5,
      style,
      horizontal = false,
      separatorVariant = 'neutral-alpha-200',
      headerSeparator = false,
      footerSeparator = false,
      ...props
    },
    ref
  ) => {
    const customRenderSeparator = () => {
      return (
        <Separator
          size='full'
          orientation={!horizontal ? 'horizontal' : 'vertical'}
          variant={separatorVariant}
          width={props.separatorWidth}
          height={props.separatorHeight}
        />
      )
    }

    return (
      <Item
        style={StyleSheet.flatten([
          ListStyles({width, ...props}).defaultListContainerStyle,
          style,
        ])}>
        <FlashList
          data={data}
          ref={ref}
          estimatedItemSize={data ? (data?.length === 0 ? 1 : data.length) : estimatedItemSize}
          ItemSeparatorComponent={renderSeparator ? () => customRenderSeparator() : undefined}
          ListHeaderComponent={headerSeparator ? () => customRenderSeparator() : undefined}
          ListFooterComponent={footerSeparator ? () => customRenderSeparator() : undefined}
          horizontal={horizontal}
          {...props}
        />
      </Item>
    )
  }
)
