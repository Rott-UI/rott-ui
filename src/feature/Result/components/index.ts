export * from './Result'
export * from './ResultScreen'
