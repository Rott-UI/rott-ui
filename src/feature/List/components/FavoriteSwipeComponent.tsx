// React Imports
import {FC} from 'react'

// Component Imports
import {Icon, Item, Label, Pressable} from '../../../components'

// Constant Imports
import {COLOURS} from '../../../constants'

// Util and Lib Imports
import {formatMessage} from '../../../utils'

interface FavoriteSwipeComponentProps {
  index: number
  favorite: boolean
  handleFavoriteChange: (itemIndex: number, itemFavorite: boolean) => void | undefined
}

export const FavoriteSwipeComponent: FC<FavoriteSwipeComponentProps> = ({
  index,
  favorite,
  handleFavoriteChange,
}) => (
  <Item row width={48}>
    <Pressable
      alignItemsCenter
      justifyContentCenter
      backgroundColor={favorite ? COLOURS.SEMANTIC_DANGER : COLOURS.SECONDARY}
      onPress={() => handleFavoriteChange && handleFavoriteChange!(index, !favorite)}>
      <Icon
        width={16}
        height={16}
        name={favorite ? 'STAR_FILL' : 'STAR'}
        variant={favorite ? 'secondary' : 'info'}
      />

      <Label textCenter color={favorite ? COLOURS.WHITE : COLOURS.BLACK} fontSize={8} marginTop={4}>
        {formatMessage(favorite ? 'LIST.ITEM.FAVORITE.REMOVE' : 'LIST.ITEM.FAVORITE.ADD')}
      </Label>
    </Pressable>
  </Item>
)
