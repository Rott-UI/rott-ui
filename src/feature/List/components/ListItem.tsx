/* eslint-disable react-native/no-inline-styles */
// React Imports
import {FC} from 'react'

// Component Imports
import {Icon, Item, Label, Pressable} from '../..'

// Util and Lib Imports
import {ListItemProps} from '../utils'
import {colourFromVariant} from '../../../utils'

// Package Imports
import Swipeable from 'react-native-gesture-handler/Swipeable'

/**
 *
 * @param icon - Solda renderlanacak icon
 * @param title - Başlık **Zorunlu**
 * @param subTitle - Alt Başlık
 * @param rightIcon - Sağda renderlanacak icon
 * @param color - Yazi Rengi
 * @param fontFamily - Yazı Fontu
 * @param onAction - Tıklandığında Alacağı Fonksiyon
 * @param backgroundColor - Item Arka Plan Rengi varsayilan degeri `green-vogue`
 * @param swipeable - Kaydirma Efekti
 * @param renderRightActions - Kaydirma Efektinde Sag Tarafta Renderlanacak Item Degerleri
 * @param renderLeftActions - Kaydirma Efektinde Sol Tarafta Renderlanacak Item Degerleri
 *
 * @returns Standart List Item Renderlar
 */
export const ListItem: FC<ListItemProps> = ({
  testID,
  icon,
  title,
  subTitle,
  rightIcon,
  titleColor = 'grey-900',
  iconColor = 'grey-900',
  rightIconColor = 'grey-900',
  fontFamily = 'Markpro-Medium',
  backgroundColor = 'grey-200',
  swipeable,
  renderRightActions,
  renderLeftActions,
  onAction,
}) => {
  const listItemContent = (
    <Pressable
      key={testID ?? `list-item-${title.toSeoFriendly()}`}
      backgroundColor={colourFromVariant(backgroundColor)}
      onPress={() => onAction && onAction!()}>
      <Item
        testID={testID ?? `list-item-${title.toSeoFriendly()}-test-id`}
        row
        alignItemsCenter
        justifyContentCenter
        height={56}
        width={390}
        paddingHorizontal={24}>
        <Item width={24}>
          {icon && (
            <Icon name={icon} width={24} height={24} variant={iconColor} noStroke mode='stroke' />
          )}
        </Item>

        <Item marginLeft={16} width={262}>
          {icon && rightIcon && (
            <Label variant={titleColor} fontSize='xl' fontFamily={fontFamily}>
              {title}
            </Label>
          )}

          {icon && !rightIcon && (
            <Label variant={titleColor} fontSize='xl' fontFamily={fontFamily}>
              {title}
            </Label>
          )}

          {!icon && !rightIcon && (
            <Label variant={titleColor} fontSize='xl' fontFamily={fontFamily} justifyContentCenter>
              {title}
            </Label>
          )}

          {subTitle && (
            <Label variant={titleColor} fontSize='lg' fontFamily={fontFamily} marginTop={8}>
              {subTitle}
            </Label>
          )}
        </Item>

        <Item width={24} marginLeft={16}>
          {rightIcon && <Icon name={rightIcon} width={24} height={24} variant={rightIconColor} />}
        </Item>
      </Item>
    </Pressable>
  )

  return (
    <>
      {swipeable && (
        <Swipeable renderRightActions={renderRightActions} renderLeftActions={renderLeftActions}>
          {listItemContent}
        </Swipeable>
      )}

      {!swipeable && {...listItemContent}}
    </>
  )
}
