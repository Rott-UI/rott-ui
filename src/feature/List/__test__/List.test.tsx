// React Imports
import React from 'react'

// React Native Imports
import {Text} from 'react-native'

// Component Imports
import {List} from '../components'

// Util and Lib Imports
import {render, waitFor} from '../../../utils'

describe('List -> Custom Component', () => {
  const customSeparatorTestId = 'custom-separator-test-id'

  const data = [
    {id: 1, name: 'Item 1'},
    {id: 2, name: 'Item 2'},
    {id: 3, name: 'Item 3'},
  ]

  it('list ilk render anında snapshot ile eşleşmeli', () => {
    const renderedList = render(
      <List
        data={data}
        estimatedItemSize={100}
        renderSeparator={true}
        separatorWidth={2}
        separatorHeight={20}
        renderItem={undefined}
      />
    )

    expect(renderedList).toMatchSnapshot()
  })

  it('separator item ekranda olmamalı', () => {
    const {queryByTestId} = render(
      <List
        data={data}
        estimatedItemSize={100}
        renderSeparator={false}
        separatorWidth={2}
        separatorHeight={20}
        renderItem={undefined}
      />
    )

    const separator = queryByTestId(customSeparatorTestId)

    expect(separator).toBeNull()
  })

  it('list componenti elemanları doğru şekilde renderlamalı', () => {
    const {getByText} = render(
      <List
        data={data}
        horizontal={false}
        numColumns={1}
        renderItem={({item}) => <Text>{item.name}</Text>}
      />
    )

    data.forEach((itemText) => {
      const itemElement = getByText(itemText.name)
      expect(itemElement).toBeDefined()
    })
  })

  it('list componenti eleman geldiğinde doğru renderlanmalı', async () => {
    const {getByText} = render(
      <List data={data} renderSeparator={true} renderItem={({item}) => <Text>{item.name}</Text>} />
    )

    for (const item of data) {
      await waitFor(() => {
        const itemText = getByText(item.name)
        expect(itemText).toBeDefined()
      })
    }
  })
})
