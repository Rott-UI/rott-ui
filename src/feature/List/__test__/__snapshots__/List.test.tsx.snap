// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List -> Custom Component list ilk render anında snapshot ile eşleşmeli 1`] = `
[
  <View
    style={
      {
        "alignItems": undefined,
        "backgroundColor": undefined,
        "bottom": undefined,
        "flex": undefined,
        "flexDirection": "column",
        "flexGrow": undefined,
        "gap": undefined,
        "height": undefined,
        "justifyContent": undefined,
        "left": undefined,
        "marginBottom": undefined,
        "marginLeft": undefined,
        "marginRight": undefined,
        "marginTop": undefined,
        "maxHeight": undefined,
        "maxWidth": undefined,
        "overflow": undefined,
        "paddingBottom": undefined,
        "paddingHorizontal": undefined,
        "paddingLeft": undefined,
        "paddingRight": undefined,
        "paddingTop": undefined,
        "paddingVertical": undefined,
        "position": undefined,
        "right": undefined,
        "top": undefined,
        "width": undefined,
        "zIndex": undefined,
      }
    }
  >
    <View
      style={
        {
          "flex": 1,
          "overflow": "hidden",
        }
      }
    >
      <RCTScrollView
        ItemSeparatorComponent={[Function]}
        applyWindowCorrection={[Function]}
        canChangeSize={true}
        contentContainerStyle={
          {
            "backgroundColor": undefined,
            "minHeight": 1,
            "minWidth": 1,
            "paddingLeft": 0,
            "paddingRight": 0,
          }
        }
        contentHeight={9}
        contentWidth={400}
        data={
          [
            {
              "id": 1,
              "name": "Item 1",
            },
            {
              "id": 2,
              "name": "Item 2",
            },
            {
              "id": 3,
              "name": "Item 3",
            },
          ]
        }
        dataProvider={
          DataProvider {
            "_data": [
              {
                "id": 1,
                "name": "Item 1",
              },
              {
                "id": 2,
                "name": "Item 2",
              },
              {
                "id": 3,
                "name": "Item 3",
              },
            ],
            "_firstIndexToProcess": 0,
            "_hasStableIds": false,
            "_requiresDataChangeHandling": false,
            "_size": 3,
            "getStableId": [Function],
            "rowHasChanged": [Function],
          }
        }
        disableRecycling={false}
        estimatedItemSize={3}
        externalScrollView={[Function]}
        finalRenderAheadOffset={250}
        forceNonDeterministicRendering={true}
        horizontal={false}
        initialOffset={0}
        initialRenderIndex={0}
        isHorizontal={false}
        layoutProvider={
          GridLayoutProviderWithProps {
            "_acceptableRelayoutDelta": 1,
            "_getHeightOrWidth": [Function],
            "_getLayoutTypeForIndex": [Function],
            "_getSpan": [Function],
            "_hasExpired": false,
            "_isHorizontal": false,
            "_lastLayoutManager": GridLayoutManager {
              "_acceptableRelayoutDelta": 1,
              "_getSpan": [Function],
              "_isGridHorizontal": false,
              "_isHorizontal": false,
              "_layoutProvider": [Circular],
              "_layouts": [
                {
                  "height": 3,
                  "type": 0,
                  "width": 400,
                  "x": 0,
                  "y": 0,
                },
                {
                  "height": 3,
                  "type": 0,
                  "width": 400,
                  "x": 0,
                  "y": 3,
                },
                {
                  "height": 3,
                  "type": 0,
                  "width": 400,
                  "x": 0,
                  "y": 6,
                },
              ],
              "_maxSpan": 1,
              "_renderWindowSize": {
                "height": 900,
                "width": 400,
              },
              "_totalHeight": 9,
              "_totalWidth": 400,
              "_window": {
                "height": 900,
                "width": 400,
              },
            },
            "_maxSpan": 1,
            "_renderWindowSize": {
              "height": 900,
              "width": 400,
            },
            "_setLayoutForType": [Function],
            "_tempDim": {
              "height": 0,
              "width": 0,
            },
            "averageWindow": AverageWindow {
              "currentAverage": 3,
              "currentCount": 1,
              "inputValues": [
                3,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
                ,
              ],
              "nextIndex": 1,
            },
            "defaultEstimatedItemSize": 100,
            "layoutObject": {
              "size": undefined,
              "span": undefined,
            },
            "props": {
              "ItemSeparatorComponent": [Function],
              "ListFooterComponent": undefined,
              "ListHeaderComponent": undefined,
              "data": [
                {
                  "id": 1,
                  "name": "Item 1",
                },
                {
                  "id": 2,
                  "name": "Item 2",
                },
                {
                  "id": 3,
                  "name": "Item 3",
                },
              ],
              "estimatedItemSize": 3,
              "horizontal": false,
              "numColumns": 1,
              "renderItem": undefined,
              "separatorHeight": 20,
              "separatorWidth": 2,
            },
            "renderWindowInsets": {
              "height": 0,
              "width": 0,
            },
            "shouldRefreshWithAnchoring": true,
          }
        }
        maxRenderAhead={750}
        numColumns={1}
        onEndReached={[Function]}
        onEndReachedThreshold={0}
        onEndReachedThresholdRelative={0}
        onItemLayout={[Function]}
        onLayout={[Function]}
        onScroll={[Function]}
        onScrollBeginDrag={[Function]}
        onSizeChanged={[Function]}
        onVisibleIndicesChanged={[Function]}
        removeClippedSubviews={false}
        renderAheadOffset={0}
        renderAheadStep={250}
        renderContentContainer={[Function]}
        renderItemContainer={[Function]}
        rowRenderer={[Function]}
        scrollEventThrottle={16}
        scrollThrottle={16}
        scrollViewProps={
          {
            "contentContainerStyle": {
              "backgroundColor": undefined,
              "minHeight": 1,
              "minWidth": 1,
              "paddingLeft": 0,
              "paddingRight": 0,
            },
            "onLayout": [Function],
            "onScrollBeginDrag": [Function],
            "refreshControl": undefined,
            "style": {
              "minHeight": 1,
              "minWidth": 1,
            },
          }
        }
        separatorHeight={20}
        separatorWidth={2}
        style={
          {
            "minHeight": 1,
            "minWidth": 1,
          }
        }
        suppressBoundedSizeException={true}
        windowCorrectionConfig={
          {
            "applyToInitialOffset": true,
            "applyToItemScroll": true,
            "value": {
              "endCorrection": 0,
              "startCorrection": 0,
              "windowShift": -0,
            },
          }
        }
      >
        <View>
          <View
            style={
              {
                "flexDirection": "column",
              }
            }
          >
            <View
              style={
                {
                  "paddingLeft": 0,
                  "paddingTop": 0,
                }
              }
            />
            <View
              style={
                [
                  undefined,
                  undefined,
                ]
              }
            />
            <AutoLayoutView
              enableInstrumentation={false}
              horizontal={false}
              onBlankAreaEvent={[Function]}
              onLayout={[Function]}
              renderAheadOffset={0}
              scrollOffset={0}
              style={
                {
                  "height": 9,
                  "width": 400,
                }
              }
              windowSize={900}
            >
              <CellContainer
                index={0}
                onLayout={[Function]}
                style={
                  {
                    "alignItems": "stretch",
                    "flexDirection": "column",
                    "left": 0,
                    "position": "absolute",
                    "top": 0,
                    "width": 400,
                  }
                }
              >
                <View
                  style={
                    {
                      "flexDirection": "column",
                    }
                  }
                />
                <View
                  style={
                    {
                      "backgroundColor": "rgba(34, 63, 70, 0.15)",
                      "bottom": undefined,
                      "flexGrow": undefined,
                      "gap": undefined,
                      "height": 39,
                      "left": undefined,
                      "marginBottom": undefined,
                      "marginLeft": undefined,
                      "marginRight": undefined,
                      "marginTop": undefined,
                      "maxHeight": 39,
                      "maxWidth": undefined,
                      "opacity": 1,
                      "overflow": undefined,
                      "paddingBottom": undefined,
                      "paddingHorizontal": undefined,
                      "paddingLeft": undefined,
                      "paddingRight": undefined,
                      "paddingTop": undefined,
                      "paddingVertical": undefined,
                      "position": undefined,
                      "right": undefined,
                      "top": undefined,
                      "width": "100%",
                      "zIndex": undefined,
                    }
                  }
                  variant="neutral-alpha-200"
                />
              </CellContainer>
              <CellContainer
                index={1}
                onLayout={[Function]}
                style={
                  {
                    "alignItems": "stretch",
                    "flexDirection": "column",
                    "left": 0,
                    "position": "absolute",
                    "top": 3,
                    "width": 400,
                  }
                }
              >
                <View
                  style={
                    {
                      "flexDirection": "column",
                    }
                  }
                />
                <View
                  style={
                    {
                      "backgroundColor": "rgba(34, 63, 70, 0.15)",
                      "bottom": undefined,
                      "flexGrow": undefined,
                      "gap": undefined,
                      "height": 39,
                      "left": undefined,
                      "marginBottom": undefined,
                      "marginLeft": undefined,
                      "marginRight": undefined,
                      "marginTop": undefined,
                      "maxHeight": 39,
                      "maxWidth": undefined,
                      "opacity": 1,
                      "overflow": undefined,
                      "paddingBottom": undefined,
                      "paddingHorizontal": undefined,
                      "paddingLeft": undefined,
                      "paddingRight": undefined,
                      "paddingTop": undefined,
                      "paddingVertical": undefined,
                      "position": undefined,
                      "right": undefined,
                      "top": undefined,
                      "width": "100%",
                      "zIndex": undefined,
                    }
                  }
                  variant="neutral-alpha-200"
                />
              </CellContainer>
              <CellContainer
                index={2}
                onLayout={[Function]}
                style={
                  {
                    "alignItems": "stretch",
                    "flexDirection": "column",
                    "left": 0,
                    "position": "absolute",
                    "top": 6,
                    "width": 400,
                  }
                }
              >
                <View
                  style={
                    {
                      "flexDirection": "column",
                    }
                  }
                />
              </CellContainer>
            </AutoLayoutView>
            <CellContainer
              index={-1}
              style={
                [
                  undefined,
                  undefined,
                ]
              }
            />
            <View
              style={
                {
                  "paddingBottom": 0,
                  "paddingRight": 0,
                }
              }
            />
          </View>
        </View>
      </RCTScrollView>
    </View>
  </View>,
  <View
    onLayout={[Function]}
    pointerEvents="box-none"
    style={
      [
        {
          "alignSelf": "center",
          "elevation": 999999,
          "flex": 0,
          "maxWidth": "100%",
          "position": "absolute",
          "zIndex": 999999,
        },
        {
          "flexDirection": "column-reverse",
          "justifyContent": "flex-start",
          "top": 30,
          "width": 750,
        },
      ]
    }
  >
    <View
      style={
        {
          "bottom": 0,
        }
      }
    >
      <RCTSafeAreaView />
    </View>
  </View>,
  <View
    onLayout={[Function]}
    pointerEvents="box-none"
    style={
      [
        {
          "alignSelf": "center",
          "elevation": 999999,
          "flex": 0,
          "maxWidth": "100%",
          "position": "absolute",
          "zIndex": 999999,
        },
        {
          "bottom": 40,
          "flexDirection": "column",
          "justifyContent": "flex-end",
          "width": 750,
        },
      ]
    }
  >
    <View
      style={
        {
          "bottom": 0,
        }
      }
    >
      <RCTSafeAreaView />
    </View>
  </View>,
]
`;
